File: GeeksForGeeks-Problems/JAVASCRIPT/JavaScript Fundamentals/Find Pattern.js
class Solution {
    findPattern(s, p) {
        // code here
        for(let i=0;i<=s.length-p.length;i++){
            let match=true;
            for(let j=0;j<p.length;j++){
                if(s[i+j]!==p[j]){
                    match=false;
                    break;
                }
            }
            if(match) return i;
        }
        return -1;
        
        // return s.indexOf(p);
    }
}

File: GeeksForGeeks-Problems/CPP/String/Find Pattern.cpp
// User function Template for C++

// function should find if the string p is
// in the string s. Return the starting index
// of the string p in s. Return -1 if the
// string doesnot exists in s.
int findPattern(string &s, string &p)
{
    // code here
    int ans = s.find(p);
    if (ans != string::npos)
        return ans;
    return -1;
}